File: algos_assignment/2nd_grade/Kim Jihyeon_ttplove2/week6/1920.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int N, M;
int arr[100001];

int binarySearch(int s, int e, int key)
{
	while (s <= e) {
		int mid = (s + e) / 2;
		if (arr[mid] > key) e = mid - 1;
		else if (arr[mid] < key) s = mid + 1;
		else return 1;
	}
	return 0;
}

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);

	cin >> N;
	for (int i = 0; i < N; i++) {
		cin >> arr[i];
	}
	sort(arr, arr + N);

	cin >> M;
	for (int i = 0; i < M; i++) {
		int x; cin >> x;
		cout << binarySearch(0, N - 1, x) << "\n";
	}
	return 0;
}

File: algos_assignment/2nd_grade/김가영_kijh30123/week3/1920.py
import sys

def binary_search(num):
    left=0
    right=len(arr)
    while left<right:
        mid=(left+right)//2
        if arr[mid]<num:
            left=mid+1
        elif arr[mid]==num:
            return 1
        else:
            right=mid
    return 0

N=int(input())
arr=list(map(int,sys.stdin.readline().split()))
M=int(input())
b=list(map(int,sys.stdin.readline().split()))

arr.sort()

for i in range(M):
    print(binary_search(b[i]))
